### PR TITLE
[BD-14] Adjust API for global phase-in of database-backed orgs with optional strictness

### PR DESCRIPTION
**Pull request by kdmccormick**
Implements part of [TNL-7425](https://openedx.atlassian.net/browse/TNL-7425).

Needed by https://github.com/edx/edx-platform/pull/25153

See the linked ADR for context.

Note: As I work on [the edx-platform side]( https://github.com/edx/edx-platform/pull/25153) of this ticket, I'm realizing I need to make [further API additions in this repo](https://github.com/edx/edx-organizations/pull/138), but those new additions shouldn't block review and merge of this.

### CODE REVIEW

**Review by ormsbee — CHANGES_REQUESTED** [review]
Minor questions/comments about backfill and naming.

File: docs/decisions/0001-phase-in-db-backed-organizations-to-all.rst [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+instances, without an ability to disable them.
+However, we do not want to force organization validation upon all Open edX instances.
+So, we will **install the organizations app by default** but also
+**make organization validation an optional feature**.
+When the organization validation is disabled,
+we will **automatically populate the organizations table** as course runs or
+content libraries are created.
+We will provide an **organization backfill management command**,
+but running it will not be required for instances that are not concerned about
+organization-related features.
```
[the remark this answers is not in the document]
Comment (kdmccormick) [reply docs/decisions/0001-phase-in-db-backed-organizations-to-all.rst:102]: Agreed. I'll update this.

File: docs/decisions/0001-phase-in-db-backed-organizations-to-all.rst [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+``ORGANIZATIONS_ENABLE_STRICTNESS``.
+
+When ``True`` ("strict mode"), the ``Organization`` model will be
+treated as an **authoritative source** of data, so creating content that references
+a non-existant organization should result in a validation error.
+
+When ``False`` ("relaxed mode"), the ``Organization`` model will be
+treated as a **downstream source** of data, so creating content that
+references a non-existant organization should cause that organization to be
+automatically created in the database.
```
[the remark this answers is not in the document]
Comment (kdmccormick) [reply docs/decisions/0001-phase-in-db-backed-organizations-to-all.rst:124]: yeah, I originally had it as `ENABLE_STRICT_ORGANIZATIONS`, but I wanted it to follow the `{PACKAGE}_{SETTING}` idiom that most Django packages follow.

anyway, I think I like `ORGANIZATIONS_AUTOCREATE` more.

**Comment by kdmccormick** [comment]
@ormsbee Ready for another pass. I addressed your comments in the fixup commits.

**Review by ormsbee — APPROVED** [review]

**Pull request merged**